File: java/oop/Polymorph.java
class Animal {
    void sound() {
        System.out.println("Animal makes a sound");
    }
}

class Dog extends Animal {
    void sound() {
        System.out.println("Dog barks");
    }
}

class Cat extends Animal {
    void sound() {
        System.out.println("Cat meows");
    }
}

public class Polymorphism {
    public static void main(String[] args) {
        Animal myAnimal = new Dog();
        myAnimal.sound(); // Outputs: Dog barks

        myAnimal = new Cat();
        myAnimal.sound(); // Outputs: Cat meows
    }
}

File: cpp/polymorph.cpp
#include <iostream>

class Animal {
public:
    virtual void speak() { // Virtual function
        std::cout << "Animal speaks" << std::endl;
    }
};

class Dog : public Animal {
public:
    void speak() override { // Override function
        std::cout << "Dog barks" << std::endl;
    }
};

int main() {
    Animal* animal = new Dog(); // Polymorphism
    animal->speak(); // Calls Dog's speak method

    delete animal;
    return 0;
}
